src/CredentialAgent.js

```diff
@@ -0,0 +1,59 @@
+import fetch from './fetch';
+import { Login } from '.';
+
+/**
+ * A credential agent that fetches the credentials from the taskcluster
+ * login service, given an access_token.
+ *
+ * Users can keep a single such credential agent and use it simultaneously
+ * with many Client instances.
+ */
+export class OIDCCredentialAgent {
```

Comment:
Each of these classes needs to be in its own file, maybe in an `agents` or `credentialAgents` directory.